docusaurus/docs/reference/ha/bootstrapping/certificates.md

```diff
@@ -0,0 +1,86 @@
+---
+
+# Controller Certificates
+
+For controllers to communicate and trust one another, they need certificates that have
+been generated with the correct attribute and relationships.
+
+## Requirements
+
+1. The certificates must have a shared root of trust
+2. The controller client and server certificates must contain a 
+   [SPIFFE ID](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-id)
+
+## Steps to Certificate Creation
+There are many ways to set up certificates, so this will just cover a recommended configuration.
+
+The primary thing to ensure is that controllers have a shared root of trust. 
+A standard way of generating certs would be as follows:
+
+1. Create a self-signed root CA
+1. Create an intermediate signing cert for each controller
+1. Create a server cert using the signing cert for each controller
+1. Create a client cert using the signing cert for each controller
+1. Make sure that the CA bundle for each server includes both the root CA and the intermediate CA
+   for that server
+
+Note that controller server and client certs must contain a SPIFFE id of the form
+
+```
+spiffe://<trust domain>/controller/<controller id>
+```
+
+So if your trust domain is `example.com` and your controller id is `ctrl1`, then your SPIFFE id
+would be:
+
+```
+spiffe://example.com/controller/ctrl1
+```
+
+**SPIFFE ID Notes:**
+
+* This ID must be set as the only URI in the `X509v3 Subject Alternative Name` field in the
+  certificate.
+* These IDs are used to allow the controllers to identify each during the mTLS negotiation.
+* The OpenZiti CLI supports creating SPIFFE IDs in your certs
+    * Use the `--trust-domain` flag when creating CAs
+    * Use the `--spiffe-id` flag when creating server or client certificates
+
+## Example
+
+Using the OpenZiti PKI tool, certificates for a three node cluster could be created as follows:
+
+```bash
+# Create the trust root, a self-signed CA
+ziti pki create ca --trust-domain ha.test --pki-root ./pki --ca-file ca --ca-name 'HA Example Trust Root'
+
+# Create the controller 1 intermediate/signing cert
+ziti pki create intermediate --pki-root ./pki --ca-name ca --intermediate-file ctrl1 --intermediate-name 'Controller One Signing Cert'
+
+# Create the controller 1 server cert
+ziti pki create server --pki-root ./pki --ca-name ctrl1 --dns localhost --dns ctrl1.ziti.example.com --ip 127.0.0.1 --server-name ctrl1 --spiffe-id 'controller/ctrl1'
+
+# Create the controller 1 server cert
+ziti pki create client --pki-root ./pki --ca-name ctrl1 --client-name ctrl1 --spiffe-id 'controller/ctrl1'
+
+# Create the controller 2 intermediate/signing cert
+ziti pki create intermediate --pki-root ./pki --ca-name ca --intermediate-file ctrl2 --intermediate-name 'Controller Two Signing Cert'
+
+# Create the controller 2 server cert
+ziti pki create server --pki-root ./pki --ca-name ctrl2 --dns localhost --dns ctrl2.ziti.example.com --ip 127.0.0.1 --server-name ctrl2 --spiffe-id 'controller/ctrl2'
```

Comment:
This seems to reflect a local dev env more than an operational scenario. We shouldn't guide the reader to generate subsequent controllers' private keys and then copy them from the place where they're issued to the nodes where they're used, and that isn't possible anyway when the private key material is non-portable from its runtime environment. 

Shouldn't we be using `ziti pki create csr` to request the subsequent intermediates on their eventual hosting environments where the private keys can be generated and never transported, i.e., where their non-portable private keys are available?